### DISCUSSION TITLE
Constraint anchor positions

### DISCUSSION THREAD

**Discussion by jhansson** [godot-jolt/godot-jolt#1018] [Questions] [answered]
Is it possible to set custom "anchor" positions for where constraints are attached to the node_a and node_b? Or is it always at their origin?

**Comment by mihe** [comment, marked as answer]
It's technically possible by creating the joints through the physics server interface (`PhysicsServer3D.joint_create`, `PhysicsServer3D.joint_make_hinge`, etc.) which takes the local-space reference transforms as arguments, but it's a bit messy, undocumented and means you'd be creating your own joint nodes from scratch.

There's no way to do it with the existing joint nodes though, no. The reference frames are always the joint's transform local to the respective body.
